packages/angular/src/plugins/plugin.ts

```diff
@@ -0,0 +1,733 @@
+import {
+  createNodesFromFiles,
+  detectPackageManager,
+  getPackageManagerCommand,
+  readJsonFile,
+  writeJsonFile,
+  type CreateNodesContextV2,
+  type CreateNodesResult,
+  type CreateNodesV2,
+  type ProjectConfiguration,
+  type Target,
+  type TargetConfiguration,
+} from '@nx/devkit';
+import { calculateHashForCreateNodes } from '@nx/devkit/src/utils/calculate-hash-for-create-nodes';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+import { getLockFileName } from '@nx/js';
+import { existsSync, readdirSync, statSync } from 'node:fs';
+import { dirname, join, relative } from 'node:path';
+import * as posix from 'node:path/posix';
+import { hashObject } from 'nx/src/devkit-internals';
+import { workspaceDataDirectory } from 'nx/src/utils/cache-directory';
+
+export interface AngularPluginOptions {}
+
+type AngularProjects = Record<
+  string,
+  Pick<ProjectConfiguration, 'projectType' | 'sourceRoot' | 'targets'>
+>;
+
+type AngularTargetConfiguration = {
+  builder: string;
+  options?: Record<string, any>;
+  configurations?: Record<string, any>;
+  defaultConfiguration?: string;
+};
+export type AngularProjectConfiguration = {
+  projectType: 'application' | 'library';
+  root: string;
+  sourceRoot?: string;
+  architect?: Record<string, AngularTargetConfiguration>;
+  targets?: Record<string, AngularTargetConfiguration>;
+};
+type AngularJson = { projects?: Record<string, AngularProjectConfiguration> };
+
+const knownExecutors = {
+  appShell: new Set(['@angular-devkit/build-angular:app-shell']),
+  build: new Set([
+    '@angular-devkit/build-angular:application',
+    '@angular/build:application',
+    '@angular-devkit/build-angular:browser-esbuild',
+    '@angular-devkit/build-angular:browser',
+    '@angular-devkit/build-angular:ng-packagr',
+  ]),
+  devServer: new Set(['@angular-devkit/build-angular:dev-server']),
+  extractI18n: new Set(['@angular-devkit/build-angular:extract-i18n']),
+  prerender: new Set(['@angular-devkit/build-angular:prerender']),
+  server: new Set(['@angular-devkit/build-angular:server']),
+  serveSsr: new Set(['@angular-devkit/build-angular:ssr-dev-server']),
+  test: new Set(['@angular-devkit/build-angular:karma']),
+};
+
+const pmc = getPackageManagerCommand();
+
+function readProjectsCache(cachePath: string): Record<string, AngularProjects> {
+  return existsSync(cachePath) ? readJsonFile(cachePath) : {};
+}
+
+function writeProjectsToCache(
+  cachePath: string,
+  results: Record<string, AngularProjects>
+) {
+  writeJsonFile(cachePath, results);
+}
+
+export const createNodesV2: CreateNodesV2<AngularPluginOptions> = [
+  '**/angular.json',
+  async (configFiles, options, context) => {
+    const optionsHash = hashObject(options);
```

Comment:
How safe is this hash if the options is empty/undefined? (because of empty interface)

Reply:
That's handled by the function. Regardless of an empty interface the options can always be undefined since plugins are designed to not require them.